### PR TITLE
Folder sharing with entire org tf update

### PR DESCRIPTION
**Pull request by JKashyap96**

### CODE REVIEW

**Comment by JKashyap96** [comment]
Will be having more commits, as e2e testing pending.

File: aws-observability-terraform/app-modules/variables.tf [hunk 1]
```diff
@@ -167,5 +167,22 @@ variable "folder_installation_location" {
...
+        EOT
+  validation {
+    condition     = can(regex("\\w+", var.sumologic_organization_id))
+    error_message = "The organization ID must contain valid characters."
+  }
+}
 
+variable "folder_share_with_org" {
+  type        = bool
+  description = "Indicates if AWSO folder should be shared with entire organization. true to enable; false to disable."
```
Comment (himsharma01) [inline aws-observability-terraform/app-modules/variables.tf:184]: AWSO to "AWS Observability:

File: aws-observability-terraform/variables.tf [hunk 2]
```diff
@@ -76,3 +76,9 @@ variable "sumologic_folder_installation_location" {
 
 }
 
+variable "sumologic_folder_share_with_org" {
+  type        = bool
+  description = "Indicates if AWSO folder should be shared (view access) with entire organization. true to enable; false to disable."
```
Comment (himsharma01) [inline aws-observability-terraform/variables.tf:81]: AWSO to "AWS Observability"
In description of this var

**Comment by himsharma01** [comment]
@JKashyap96 @npande - according to me folder sharing variables should be part of main.tf instead of main.auto.tfvars. Please suggest.

File: aws-observability-terraform/main.auto.tfvars [hunk 3]
```diff
@@ -5,4 +5,5 @@ sumologic_access_id       = "<YOUR SUMO ACCESS ID>"    # Please replace <YOUR SU
 sumologic_access_key      = "<YOUR SUMO ACCESS KEY>"   # Please replace <YOUR SUMO ACCESS KEY> (including brackets) with your Sumo Logic Access KEY.
 sumologic_organization_id = "<YOUR SUMO ORG ID>"       # Please replace <YOUR SUMO ORG ID> (including brackets) with your Sumo Logic Organization ID.
 aws_account_alias         = "<YOUR AWS ACCOUNT ALIAS>" # Please replace <YOUR AWS ACCOUNT ALIAS> with an AWS account alias for identification in Sumo Logic Explorer View, metrics and logs.
-sumologic_folder_installation_location = "<YOUR APP FOLDER LOCATION>" # Please replace <YOUR APP FOLDER LOCATION> with your preferred folder installation location, either of "Personal Folder" or "Admin Recommended Folder". Default is "Personal Folder".
\ No newline at end of file
+sumologic_folder_installation_location = "<YOUR APP FOLDER LOCATION>" # Please replace <YOUR APP FOLDER LOCATION> with your preferred folder installation location, either of "Personal Folder" or "Admin Recommended Folder". Default is "Personal Folder".
+sumologic_folder_share_with_org = true #Please replace true with false if you don't want to share content with the entire org.
```
Comment (himsharma01) [inline aws-observability-terraform/main.auto.tfvars:9]: @JKashyap96 - We have default values of these variables in variables.tf and any overriding is recommended to be done from main.tf. So we can safely remove these lines.

File: aws-observability-terraform/app-modules/main.tf [hunk 4]
```diff
@@ -14,6 +14,18 @@ resource "sumologic_monitor_folder" "monitor_folder" {
   description = "This folder contains all the monitors for AWS Observability solution."
 }
 
+resource "sumologic_content_permission" "share_with_org" {
```
Comment (himsharma01) [inline aws-observability-terraform/app-modules/main.tf:17]: @JKashyap96 Please add comment for this resource

File: aws-observability-terraform/app-modules/variables.tf [hunk 5]
```diff
@@ -167,5 +167,22 @@ variable "folder_installation_location" {
   default = "Personal Folder"
 }
 
+variable "sumologic_organization_id" {
```
Comment (himsharma01) [inline aws-observability-terraform/app-modules/variables.tf:170]: Please add comment "Org ID is required in App module for sharing App Folder with Sumo Organisation"

Comment (JKashyap96) [reply aws-observability-terraform/app-modules/variables.tf:170]: Done

File: aws-observability-terraform/main.auto.tfvars (hunk 3 above)
Comment (JKashyap96) [reply aws-observability-terraform/main.auto.tfvars:9]: Okay

**Review by himsharma01 — APPROVED** [review]

**Review by himsharma01 — APPROVED** [review]
LGTM

**Pull request merged**